src/core/jpeg_stream.js

```diff
@@ -53,8 +43,24 @@ class JpegStream extends DecodeStream {
   }
 
   readBlock() {
+    return this.decodeImage();
```

Comment:
Why the `return` here, since the `readBlock` method isn't expected to return any data?